src/protocols/Screencopy.cpp

```diff
@@ -424,11 +422,5 @@ void CScreencopyProtocol::onOutputCommit(CMonitor* pMonitor) {
 
         f->client->lastFrame.reset();
         ++f->client->frameCounter;
-
-        framesToRemove.push_back(f);
-    }
-
-    for (auto const& f : framesToRemove) {
```

Comment:
removing this mechanic will segfault, the ONLY change should be the SP -> WP in the header

Reply:
only changing
<details>
<summary>SP -> WP (dropdown has diff)</summary>

```diff
diff --git a/src/protocols/Screencopy.cpp b/src/protocols/Screencopy.cpp
index 63024541..818dda98 100644
--- a/src/protocols/Screencopy.cpp
+++ b/src/protocols/Screencopy.cpp
@@ -148,7 +148,7 @@ void CScreencopyFrame::copy(CZwlrScreencopyFrameV1* pFrame, wl_resource* buffer_
 
     buffer = PBUFFER->buffer;
 
-    PROTO::screencopy->m_vFramesAwaitingWrite.emplace_back(self);
+    PROTO::screencopy->m_vFramesAwaitingWrite.emplace_back(self.lock());
 
     g_pHyprRenderer->m_bDirectScanoutBlocked = true;
     if (overlayCursor && !lockedSWCursors) {
diff --git a/src/protocols/Screencopy.hpp b/src/protocols/Screencopy.hpp
index cbc56edb..beb3a491 100644
--- a/src/protocols/Screencopy.hpp
+++ b/src/protocols/Screencopy.hpp
@@ -54,7 +54,7 @@ class CScreencopyFrame {
 
     bool                  good();
 
-    SP<CScreencopyFrame>  self;
+    WP<CScreencopyFrame>  self;
     WP<CScreencopyClient> client;
 
   private:
```

</details>

causes the resource to be deleted early
<pre>
[3152396.668] {Default Queue} zwlr_screencopy_frame_v1#20.buffer(1, 1920, 1080, 7680)
[3152396.671] {Default Queue} zwlr_screencopy_frame_v1#20.linux_dmabuf(875713112, 1920, 1080)
[3152396.674] {Default Queue} zwlr_screencopy_frame_v1#20.buffer_done()
[3152396.678] {Default Queue}  -> zwlr_screencopy_frame_v1#20.copy_with_damage(wl_buffer#6)
[3152450.953] {Display Queue} wl_display#1.delete_id(20)          <b><- wtf, why is it deleted early???</b>
[3152450.970] {Default Queue} zwlr_screencopy_frame_v1#20.flags(0)
[3152450.973] {Default Queue} zwlr_screencopy_frame_v1#20.damage(0, 0, 1920, 1080)
[3152450.977] {Default Queue} zwlr_screencopy_frame_v1#20.ready(0, 12113, 96949607)
[3152450.996] {Default Queue}  -> zwlr_screencopy_frame_v1#20.destroy()
</pre>

maybe its a ref counting issue when `framesToRemove.push_back(f)` taking wp of sp? idk

Reply:
nuke the destroyresource calls altogether

Reply:
but removing the `framesToRemove` mechanism doesn't balloon the size of `m_vFrames` and `m_vFramesAwaitingWrite`

presumably because the client sends destroy event for the frames?

Reply:
ye client sends destroy and in destroyResource we clean up the frames

actually right. Retain the machanism but instead of destroyResource erase it from awaitingwrite and thats it

Reply:
generally this system fucking sucks. Wait, let me rewrite this.

Reply:
also toplevel export has the same

Reply:
holy fuck how did i not realize destroyResource deleting it from m_vframes was doing it lol

Reply:
I mean, the function name is quite descriptive xd